crates/node/src/service/rpc/middleware.rs

```diff
@@ -267,18 +262,29 @@ async fn add_rpc_version_to_method(req: &mut hyper::Request<Body>) -> Result<(),
     let version = RpcVersion::from_request_path(&path)?;
 
     let whole_body = hyper::body::to_bytes(req.body_mut()).await?;
-    let mut json: Value = serde_json::from_slice(&whole_body)?;
-
-    if let Some(method) = json.get_mut("method").as_deref().and_then(Value::as_str) {
-        let new_method = format!("starknet_{}_{}", version.name(), method.strip_prefix("starknet_").unwrap_or(method));
+    let json: Value = serde_json::from_slice(&whole_body)?;
```

Comment:
i really don't like the fact that we're deserializing and reserializing the body like that :pensive: this has a perf cost, especially with serde_json::Value
but that isnt the scope of this PR i guess..

Reply:
Ya we can do that in a new PR probably, I can create an issue if you like. Do you've any specific suggestions?

Reply:
i think it would require us to handle versions very differently

the problem i have with parsing requests to a serde_json::Value is that it looks like this `https://docs.rs/serde_json/latest/serde_json/enum.Value.html`
it does one heap allocation per object, string and array, recursively for the whole request and it's really unnecessary
parsing a concrete struct is much cheaper since nested structs and enums only require one allocation as a whole unless there's a Box or Vec or such in between, so it's not that we're deserializing it twice for nothing, we're actually deserializing it first in the most inefficient way first and then deserializing it again in an okay form

Reply:
let's make an issue, right